opencti-platform/opencti-front/src/private/Root.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import * as R from 'ramda';
+import { graphql, useLazyLoadQuery, Variables } from 'react-relay';
+import CssBaseline from '@mui/material/CssBaseline';
+import { StyledEngineProvider } from '@mui/material/styles';
+import { ConnectedIntlProvider } from '../components/AppIntlProvider';
+import { ConnectedThemeProvider } from '../components/AppThemeProvider';
+import Index from './Index';
+import { UserContext } from '../utils/Security';
+import { Settings, User } from '../generated/graphql';
+
+const rootPrivateQuery = graphql`
+  query RootPrivateQuery {
+    me {
+      id
+      name
+      lastname
+      language
+      theme
+      user_email
+      theme
+      capabilities {
+        name
+      }
+    }
+    settings {
+      platform_map_tile_server_dark
+      platform_map_tile_server_light
+      platform_feature_flags {
+        id
+        enable
+      }
+      platform_modules {
+        id
+        enable
+      }
+      ...AppThemeProvider_settings
+      ...AppIntlProvider_settings
+    }
+    about {
+      version
+    }
+  }
+`;
+
+type SettingsFlag = {
+  id?: string;
+  enable?: boolean;
+};
+
+type Data = {
+  me: User;
+  settings: Settings;
+};
+
+const isFeatureEnabled = (settings: Settings, id: string) => {
+  const flags = settings.platform_feature_flags || [];
+  const feature: null | undefined | SettingsFlag = R.find(
+    (f: { id: string; enable: boolean }) => f.id === id,
+    flags,
+  );
+  return feature !== undefined && feature.enable === true;
+};
+const isModuleEnabled = (settings: Settings, id: string) => {
+  const modules = settings.platform_modules || [];
+  const module: null | undefined | SettingsFlag = R.find(
+    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+    (f: { id: any; enable: boolean }) => f.id === id,
+    modules,
+  );
+  return module !== undefined && module.enable === true;
+};
+const buildHelper = (settings: Settings) => ({
+  isModuleEnable: (id: string) => isModuleEnabled(settings, id),
+  isRuleEngineEnable: () => isModuleEnabled(settings, 'RULE_ENGINE'),
+  isFeatureEnable: (id: string) => isFeatureEnabled(settings, id),
+  isRuntimeFieldEnable: () => isFeatureEnabled(settings, 'RUNTIME_SORTING'),
+});
+
+const Root = () => {
+  const variables: Variables = [];
+  const dataResponse: Data = useLazyLoadQuery(rootPrivateQuery, variables) as Data;
```

Comment:
I did try this, however, I still don't think it will work, because it can still be `unknown` due to the fact that `useLazyLoadQuery` is `async` and so `unknown` until resolved.